phoenix-core/src/main/java/org/apache/phoenix/coprocessor/IndexRebuildRegionScanner.java

```diff
@@ -489,6 +490,42 @@ private boolean isDeleteFamily(Mutation mutation) {
         }
         return getMutationsWithSameTS(put, del);
     }
+
+    private void updateUnverifiedIndexRowCounters(Put actual,
+                                                  IndexToolVerificationResult.PhaseResult verificationPhaseResult) {
+        byte[] dataKey = indexMaintainer.buildDataRowKey(new ImmutableBytesWritable(actual.getRow()), viewConstants);
+        Pair<Put, Delete> putDeletePair = dataKeyToMutationMap.get(dataKey);
+        Put put = putDeletePair.getFirst();
+        if (put == null) {
+            return;
+        }
+        Delete del = putDeletePair.getSecond();
+        if (del != null && getMaxTimestamp(del) >= getTimestamp(put)) {
+            return;
+        }
+        byte[] indexRowKey = getIndexRowKey(indexMaintainer, put);
+        if (Bytes.compareTo(actual.getRow(), 0, actual.getRow().length,
+                indexRowKey, 0, indexRowKey.length) != 0) {
+            return;
+        }
+        List<Cell> cellList = actual.get(indexMaintainer.getEmptyKeyValueFamily().copyBytesIfNecessary(),
+                indexMaintainer.getEmptyKeyValueQualifier());
+        Cell cell = (cellList != null && !cellList.isEmpty()) ? cellList.get(0) : null;
+        if (cell == null) {
+            verificationPhaseResult.setUnknownIndexRowCount(verificationPhaseResult.getUnknownIndexRowCount() + 1);
```

Comment:
does this mean no empty CF?

Reply:
Yes. Is not it clear?

Reply:
I had to read a few lines of code above to understand that, but I also haven't worked on it. Maybe a comment will ease quick code surfing.

Reply:
Ok. I added a bunch of comments